shared/middleware/tables_api.rb

```diff
@@ -87,6 +103,10 @@ class TablesApi < Sinatra::Base
   delete %r{/v3/(shared|user)-tables/([^/]+)/([^/]+)/(\d+)$} do |endpoint, channel_id, table_name, id|
     dont_cache
     TableType.new(channel_id, storage_id(endpoint), table_name).delete(id.to_i)
+
+    limits = TableLimits.new(@@redis, endpoint, channel_id, table_name)
+    limits.decrement_row_count
```

Comment:
I don't think the delete-delete race you mentioned in comments will be a problem, since the second `delete` call earlier will throw a 404 if the record has already been deleted. Even though mis-counting wouldn't be too bad, it may be worth adding a test (or just a comment saying the order here is important) to make sure we keep getting this for free.